src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TagHelperDescriptorComparerTest.cs

```diff
@@ -47,44 +47,6 @@ public void GetHashCode_SameTagHelperDescriptors_HashCodeMatches()
         Assert.Equal(hashCode1, hashCode2);
     }
 
```

Comment:
Interestingely, this used to be `GetHashCode_FQNAndNameTagHelperDescriptors_HashCodeDoesNotMatch`, maybe we can revert it back? But not necessary.

https://github.com/dotnet/razor/commit/44a2657f0dec50bc531f1d27b482057055a7ef92#diff-6e0a89ae399c8637a37b4e0de5ae33ee4f3ccff4f97bf3e4caee25e3c35e8b64L56

Same for the removed `BoundAttributeDescriptorTest.cs`, it used to be `BoundAttributeDescriptor_HashDoesNotChangeWithType`.

Reply:
With the current design, this really isn't possible. Hash codes are now computed from their checksums. However, I think it's very weird anyway, which is why I removed the tests. Why would tag helpers encourage hash code collision for FQN and simple tag helpers when Equals would always return false? If we need that behavior, a custom equality comparer should be used like tooling does [here](https://github.com/dotnet/razor/blob/1799f2feac84180021b6b697cba0c095f643a18f/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ShortNameToFullyQualifiedComparer.cs#L9). IMO, we shouldn't bake weird hash code behavior into the objects themselves. Especially because they probably don't work as expected due to Equals returning false.